### PR TITLE
Add tsdoc package documentation

### PR DESCRIPTION
**Pull request by achrinza**
Loosely related: #4697

This PR is part of a bigger incremental doc improvement.

As a starting point, the package documentation were, for the most part, copy-&-pasted from the respective `README` files.

**Why this (and future tsdoc) PRs?**

1. Improve documentation
    * Improving the tsdocs would free up the main docs from having to state the fundamental purposes of each components.
    *  The main docs can be updated to be more "guided" for new lb4 users.
    * The main docs can instead point readers to the tsdocs as reference.
2. Improve reference docs
    * For those who need a quick reference point, the tsdocs can be useful.

**Why this PR?**
This PR fixes the emptiness this page:

image

* ^ This page is usually the first page seen by users looking for a reference doc
* Through this PR, the users will have a clearer idea on the purposes of the different packages.

**Self-TODO**
- [x] Add package docs for extensions
- [x] Check for errata
- [x] Fix code linting issues

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/authorization/src/index.ts [hunk 1]
```diff
@@ -3,6 +3,15 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+/**
+ * A LoopBack 4 component for authorization support (Role based, Permission based, Vote based)
+ *
+ * @remarks
+ * To read on key building blocks read through {@link https://loopback.io/doc/en/lb4/Loopback-component-authorization.html |LoopBack authorization docs}
```
Comment (raymondfeng) [inline packages/authorization/src/index.ts]: Please wrap the long line into multiple ones.

File: packages/context/src/index.ts [hunk 2]
```diff
@@ -3,6 +3,24 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+/**
+ * This module provides facilities to manage artifacts and their dependencies using {@link Context} in your Node.js applications. It can be used independent of the LoopBack framework.
+ *
+ * @remarks
+ * This package exposes TypeScript/JavaScript APIs and decorators to register artifacts, declare dependencies, and resolve artifacts by keys. The {@link Context} also serves as an IoC container to support dependency injection.
+ * Context and Binding are the two core concepts. A context is a registry of bindings and each binding represents a resolvable artifact by the key.
+
```
Comment (achrinza) [inline packages/context/src/index.ts]: Errata

**Comment by raymondfeng** [comment]
@achrinza Maybe it makes sense to squash all commits into one?

File: packages/core/src/index.ts [hunk 3]
```diff
@@ -3,9 +3,20 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+/**
+ * A moudle that provides the foundation for your LoopBack app.
```
Comment (derdeka) [inline packages/core/src/index.ts]: `moudle` ?

File: packages/core/src/index.ts [hunk 4]
```diff
@@ -3,9 +3,20 @@
...
+ *
+ * @remarks
+ * Unlike previous versions, it no longer contains the implementation for
```
Comment (derdeka) [inline packages/core/src/index.ts]: Mention a specific version might be better.

**Comment by achrinza** [comment]
@raymondfeng Good idea, I've squashed all the `add packageDocumentation` into a stub commit.

**Comment by achrinza** [comment]
@derdeka For the most part, these were copy&pasted from the `README`s. Though I agree, it should be consistent. I'll re-review and amend accordingly.

File: packages/tsdocs/src/index.ts [hunk 5]
```diff
@@ -4,19 +4,19 @@
...
  * {@link https://github.com/strongloop/loopback-next | loopback-next} monorepo
  * managed by {@link https://github.com/lerna/lerna | Lerna}.
  *
+ * @remarks
  * It's built on top of {@link https://api-extractor.com | Microsoft API Extractor}:
  *
  * - {@link https://github.com/Microsoft/web-build-tools/tree/master/apps/api-extractor | api-extractor}
  * - {@link https://github.com/Microsoft/web-build-tools/tree/master/apps/api-documenter | api-documenter}
  *
+ * @packageDocumentation
```
Comment (raymondfeng) [inline packages/tsdocs/src/index.ts]: @achrinza Can you explain why we need to add `@remarks` here and push down `@packageDocumentation` to the bottom?

Comment (achrinza) [reply packages/tsdocs/src/index.ts]: Docs above `@remarks` are [used as a one-liner summary](https://api-extractor.com/pages/tsdoc/tag_remarks/). Multi-line summaries don't seem to be supported, not even by the [Loopack API Docs](https://loopback.io/doc/en/lb4/apidocs.authentication.html#interfaces) (see: `UserService` interface)

`@packageDocumentation` is a modifier. It's at the bottom to follow the format used by the [docs](https://api-extractor.com/pages/tsdoc/tag_packagedocumentation/).

Comment (raymondfeng) [reply packages/tsdocs/src/index.ts]: Sounds good. My understanding is that `@packageDocumentation` can be anywhere in the first comment of the file.

File: packages/core/src/index.ts [hunk 6]
```diff
@@ -3,9 +3,21 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+/**
+ * The foundation of a LoopBack app.
+ *
+ * @remarks
+ * Unlike LoopBack 2 and 3, it no longer contains the implementation for
```
Comment (emonddr) [inline packages/core/src/index.ts]: @raymondfeng , do we want a different remark for this one...seems we'd want  a different one that doesn't involve the sentence `..it no longer contains...` . Just my 2 cents.

**Comment by dhmlau** [comment]
@achrinza, could you please address feedback from @emonddr? Thanks.

File: packages/core/src/index.ts (hunk 6 above)
Comment (achrinza) [reply packages/core/src/index.ts]: I've removed that line of remark.

**Comment by achrinza** [comment]
@dhmlau Thanks for re-raising the issue, it slipped from my radar 🙇; I've applied the feedback. WDYT?

**Comment by dhmlau** [comment]
@achrinza, thanks. Could you please also take care of the commit linting error? Thanks! 

@emonddr @raymondfeng @derdeka, may I get your review on this PR again please?

**Pull request merged**